### PR TITLE
replaced noaa with the new json urls

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunpy/issues/4339

### CODE REVIEW

**Comment by nabobalis** [comment]
The dev figure tests are failing and I am unable to build the tox env locally.

**Comment by nabobalis** [comment]
I forgot to check the examples. I will fix that.

File: sunpy/timeseries/sources/noaa.py [hunk 1]
```diff
@@ -154,6 +177,42 @@ def _parse_file(cls, filepath):
...
+        with open(filepath) as fp:
+            fp.seek(0)
+            data = pd.read_json(fp.read())
+
+        rename = {'ssn': 'mean monthly S.I.D.C. sunspot number',
+                  'smoothed_ssn': 'smoothed S.I.D.C. sunspot number',
+                  'observed_swpc_ssn': 'mean monthly SWPC/SWO sunspot number',
+                  'smoothed_swpc_ssn': 'smoothed SWPC/SWO sunspot number',
+                  'f10.7': 'mean monthly 10.7cm radio flux',
+                  'smoothed_f10.7': 'smoothed 10.7cm radio flux'}
```
Comment (Cadair) [inline sunpy/timeseries/sources/noaa.py:199]: These column names are long. I don't really have a strong opinion but it leads to a lot of typing when accessing them?

File: changelog/4340.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Data products from `~sunpy.net.dataretriever.NOAAIndicesClient` and `~sunpy.net.dataretriever.NOAAPredictClient` update to download the new JSON files.
```
[the remark this answers is not in the document]
Comment (dstansby) [reply changelog/4340.bugfix.rst:1]: 👍 I have a better (but maybe not complete) version locally

**Comment by AthKouloumvakos** [comment]
Hi @nabobalis and others! I wonder if it would be an interesting feature/extension for the nooa TimeSeries class, to can handle also other json files (with t.s.) from https://services.swpc.noaa.gov/json/. For example https://services.swpc.noaa.gov/json/goes/ (/primary/xrays-1-day.json) contain json timeseries for soft x-rays provided in real time (contrary to the goes.py class where the fits files are available after 1-2 days). I think this may be something interesting for the sunpy user to have. I attach a simple example of how I process and plot this xrays-1-day.json file, similarly it works for for the other xrays-?-day.json files. Something similar could be tailored to the noaa class, right?
[GOES_JSON_FLARE.zip](https://github.com/sunpy/sunpy/files/4898400/GOES_JSON_FLARE.zip)

**Comment by nabobalis** [comment]
Hi @AthKouloumvakos can you open an issue about that please?

File: sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json [hunk 3]
```diff
@@ -26,13 +26,14 @@
     "sunpy.map.tests.test_plotting.test_rectangle_aia171_width_height": "455246d338564add5be94c1dc0ff5740a90d21d8ef7da9f392ba3a01237ef5b8",
     "sunpy.timeseries.tests.test_timeseriesbase.test_esp_peek": "b9a0d6b7c14fe6175e4875385f3ccfb2ddffdea092792997db30066357b7ac68",
     "sunpy.timeseries.tests.test_timeseriesbase.test_eve_peek": "fe84a95c9943554660acdf5202cf68ffc13333c5db29cf5ebb1d4f296043c882",
-    "sunpy.timeseries.tests.test_timeseriesbase.test_fermi_gbm_peek": "a0addeae9ec0eda509c976e9058b2d90fa748b2c3d4a64df43b3a064d85f34d9",
+    "sunpy.timeseries.tests.test_timeseriesbase.test_fermi_gbm_peek": "5cbe52c6cd266ff85e599390979a5c431304212bd38dd112b0eaf4be31488e44",
```
Comment (dstansby) [inline sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json:29]: Don't merge with this changed!! See https://github.com/sunpy/sunpy/issues/4342

File: sunpy/timeseries/sources/noaa.py (hunk 1 above)
Comment (dstansby) [reply sunpy/timeseries/sources/noaa.py:199]: I'd be pro not renaming, and making sure there's a link to the external dataset description (that hopefully explains what these are) in the sunpy docs.

Comment (nabobalis) [reply sunpy/timeseries/sources/noaa.py:199]: But are they the same as before?

Comment (dstansby) [reply sunpy/timeseries/sources/noaa.py:199]: Sorry I didn't realise, that's definitely a good reason to keep the long names (for backcompat)!

Comment (nabobalis) [reply sunpy/timeseries/sources/noaa.py:199]: I'm asking, I don't know. I am happy to keep them the same or map them but I don't want to do it blindly.

Comment (dstansby) [reply sunpy/timeseries/sources/noaa.py:199]: I think we should keep them the same, so users don't have to change their code to update the names they are indexing the data with.

File: sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json (hunk 3 above)
Comment (nabobalis) [reply sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json:29]: Should I undo these changes and leave these tests failing for now? I would rather not wait on fixing the dev figure tests.

File: sunpy/timeseries/sources/noaa.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/timeseries/sources/noaa.py:199]: I have renamed both sets back to normal.

File: sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json (hunk 3 above)
Comment (Cadair) [reply sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json]: yeah from the issue looks like this should be reverted if it hasn't been already?

File: changelog/4340.bugfix.rst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Data products from `~sunpy.net.dataretriever.NOAAIndicesClient` and
+`~sunpy.net.dataretriever.NOAAPredictClient` have been updated to download
+new JSON files. The old text files which the data used to come in no longer
+exist.
+
+Both `sunpy.timseries.sources.NOAAIndicesTimeSeries` and
+`sunpy.timseries.sources.NOAAPredictIndicesTimeSeries` have been updated to
+support the new JSON files. Loading the old text files is still supported,
+but support for this will be removed in a future version of sunpy.
```
Comment (Cadair) [inline changelog/4340.bugfix.rst:9]: Add something about the differences in the columns?

File: sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json (hunk 3 above)
Comment (nabobalis) [reply sunpy/tests/figure_hashes_mpl_dev_ft_261_astropy_dev.json]: I have reverted it.

File: changelog/4340.bugfix.rst (hunk 4 above)
Comment (nabobalis) [reply changelog/4340.bugfix.rst:9]: I can add it.

Comment (nabobalis) [reply changelog/4340.bugfix.rst:12]: Updated.

**Pull request merged**